### PR TITLE
circleci: build docker images

### PR DESCRIPTION
**Pull request by asdwsda**
- [ ] Disable autobuild on dockerhub before merge

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -94,6 +120,33 @@ workflows:
...
+          name: Publish master docker image
+          executor: docker/machine
+          context: dockerhub
+          image: banzaicloud/istio-operator
+          tag: master
+          requires:
+            - build
+          filters:
+            branches:
+              only: master
```
Comment (Laci21) [inline .circleci/config.yml:133]: We don't use `master`, I think this is not needed.

**Comment by Laci21** [comment]
> Are the specific tags pushed from somewhere?

Ah, I guess they're pushed from here: https://github.com/banzaicloud/istio-operator/pull/418/files#diff-1d37e48f9ceff6d8030570cd36286a61R12 nvm

File: .circleci/config.yml (hunk 1 above)
Comment (khernyo) [reply .circleci/config.yml:133]: I think this job could be run for all release-* branches instead.
I'm not sure what would be a good docker tag, though. E.g. release-1.5 is not a good tag. Maybe dev-1.5?

Comment (Laci21) [reply .circleci/config.yml]: I don't think that's needed, imho there is no need to push new dev docker images on every single release branch push, it's just enough on tags.

**Pull request merged**